doc/nrf/ug_radio_fem.rst

```diff
@@ -62,12 +62,80 @@ Before you add the devicetree node in your application, complete the following s
+
+To perform the split differently (for example, to compensate for external conditions), you can use a FEM power model, either using one of the built-in ones or providing your own custom model.
+
+To use FEM power models, set the :kconfig:option:`CONFIG_MPSL_FEM_POWER_MODEL` Kconfig option to ``y``  and either select one of the built-in models or provide a custom model, as described in the following chapters.
+
+Using nRF21540 GPIO SPI built-in power model (Experimental)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. note::
+   The software maturity level of this feature is :ref:`Experimental <software_maturity>`.
+   The feature can be used for development, but it is not recommended for production.
+   This feature is incomplete in functionality or verification and can be expected to change in future releases.
+   The feature is made available in its current state, but the design and interfaces can change between release tags.
+   The feature is also labeled as ``EXPERIMENTAL`` in Kconfig files to indicate this status.
+
+To use this model, set :kconfig:option:`CONFIG_MPSL_FEM_POWER_MODEL` and :kconfig:option:`CONFIG_MPSL_FEM_POWER_MODEL_NRF21540_USE_BUILTIN` to ``y``.
+
+This feature uses a model to compensate the FEM gain for the following external conditions:
+
+* Temperature
+* FEM supply voltage
+* Carrier frequency
+* FEM input power.
+
+The model assumes that the FEM supply voltage is constant.
+To provide the value of this voltage to the MPSL subsystem, use the :kconfig:option:`CONFIG_MPSL_FEM_POWER_VOLTAGE` option.
+
+Adding custom power models
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If the way MPSL splits the TX power into components does not meet your requirements, or if you wish to implement a custom compensation model, you can provide one as follows:
+
+1. Set :kconfig:option:`CONFIG_MPSL_FEM_POWER_MODEL` to ``y``
+#. Provide an implementation of the ``mpsl_fem_power_model_to_use_get()`` function.
+   This function should return a pointer to a variable of the type ``mpsl_fem_power_model_t`` which contains pointers to the model's callbacks.
+#. Mandatorily implement the model's ``fetch`` callback (details explained below).
+#. Optionally implement the model's ``init`` callback (details explained below).
+   If no ``init`` callback is provided, pass ``NULL`` as the pointer to the callback.
+#. You can also optionally extend the ``MPSL_FEM_POWER_MODEL_CHOICE`` Kconfig choice with an option to select your custom model, for example, if you want to test multiple custom models.
+
+The ``init`` callback is called by MPSL once, after FEM configuration finishes.
+Calibration data (acquired from FEM internal registers, Kconfig options, and devicetree files) is passed to this function using a parameter of the ``mpsl_fem_calibration_data_t`` type.
+The meaning of the calibration data stored in this parameter is implementation-specific.
+For details, see the ``mpsl_fem_calibration_data_t`` type documentation.
```

Comment:
Where is this "mpsl_fem_calibration_data_t" documentation located?

Reply:
This will come when MPSL is deployed to sdk-nrfxlib, draft PR with the deploy https://github.com/nrfconnect/sdk-nrfxlib/pull/815 (MPSL sources still not merged in upstream).